src/test/java/no/unit/nva/doi/FetchDoiMetadataTest.java

```diff
@@ -0,0 +1,114 @@
+package no.unit.nva.doi;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.MalformedParametersException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+
+import static org.junit.Assert.*;
+
+public class FetchDoiMetadataTest {
+
+    private transient FetchDoiMetadata fetchDoiMetadata;
+
+    @Before
+    public void setUp() {
+        fetchDoiMetadata = new FetchDoiMetadata();
+    }
+
+    @Test
+    public void successfulResponse() {
+        String url = "https://doi.org/10.1093/afraf/ady029";
+        GatewayResponse result = (GatewayResponse) fetchDoiMetadata.handleRequest(url, null);
+        assertEquals(200, result.getStatusCode());
+        assertEquals(result.getHeaders().get("Content-Type"), "application/json");
+        String content = result.getBody();
+        assertNotNull(content);
+        assertTrue(content.contains(url));
+        assertTrue(content.contains("The political economy of banking in Angola"));
+    }
+
+    @Test
+    public void testBadRequestResponse() {
+        String url = "htps://doi.org/10.1093/afraf/ady029";
+        GatewayResponse result = (GatewayResponse) fetchDoiMetadata.handleRequest(url, null);
+        assertEquals(400, result.getStatusCode());
+        assertEquals(result.getHeaders().get("Content-Type"), "application/json");
+        String content = result.getBody();
+        assertNotNull(content);
+        assertEquals("{\"error\": \"unknown protocol: htps\"}", content);
+    }
+
+    @Test
+    public void testServerTimeoutResponse() {
+        fetchDoiMetadata.setExternalServiceTimeout(10);
+        String url = "https://doi.org/10.1093/afraf/ady029";
+        GatewayResponse result = (GatewayResponse) fetchDoiMetadata.handleRequest(url, null);
+        assertEquals(503, result.getStatusCode());
+        assertEquals(result.getHeaders().get("Content-Type"), "application/json");
+        String content = result.getBody();
+        assertNotNull(content);
+        assertEquals("{\"error\": \"connect timed out\"}", content);
+    }
+
+    @Test
+    public void test_getValidURI_valid() throws UnsupportedEncodingException, MalformedURLException, URISyntaxException {
+        String url = "https://doi.org/10.1093/afraf/ady029";
+        assertEquals(url, fetchDoiMetadata.getValidURI(url, 1024));
+    }
+
+    @Test(expected = URISyntaxException.class)
+    public void test_getValidURI_invalid() throws UnsupportedEncodingException, MalformedURLException, URISyntaxException {
```

Comment:
Do we test the expected message and logging?

Reply:
removed logging in next refactoring allready